File: src/main/java/ru/romanow/serialization/model/PublicData.java
package ru.romanow.serialization.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;

@Data
@AllArgsConstructor
@NoArgsConstructor
@XmlAccessorType(XmlAccessType.FIELD)
public class PublicData {

    @XmlElement(name = "key")
    private String key;

    @XmlElement(name = "data")
    private String data;
}

File: src/main/java/ru/romanow/serialization/model/PublicData.kt
package ru.romanow.serialization.model

import jakarta.xml.bind.annotation.XmlAccessType
import jakarta.xml.bind.annotation.XmlAccessorType
import jakarta.xml.bind.annotation.XmlElement

@XmlAccessorType(XmlAccessType.FIELD)
data class PublicData(
    @XmlElement(name = "key")
    var key: String? = null,

    @XmlElement(name = "data")
    var data: String? = null
)
